### PR TITLE
Add MTL Trainer

### PR DESCRIPTION
**Pull request by bhancock8**
This PR includes:
- (Primary) Adds the Trainer class which operates on a model and dataloaders
- A little cleanup of other pieces that touch training (see commit messages). This includes adjusting the defaults so that by default no logs or checkpoints are written (so users' workspaces aren't spammed with logs if they don't want them)

**Test plan**
Unit tests build a single-task model and multi-task model and confirms that for both, the initial accuracy is below a threshold and final accuracy is above a threshold, confirming both that the model trained and that we're successfully evaluating/returning scores. We set random seeds so that each run is deterministic (i.e., there won't be rare outlier runs that spontaneously fail the unit test).

### CODE REVIEW

**Comment by ajratner** [comment]
@bhancock8 tag me again when tests pass

**Comment by bhancock8** [comment]
Ah, one sec. The commit with the schedulers didn't make. Will update and re-ping.

**Comment by bhancock8** [comment]
Good to go now @ajratner!

File: snorkel/mtl/modules/utils.py [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-F.softmax can be used directly.
 """
 
 
 import torch.nn.functional as F
 
 
 def ce_loss(module_name, outputs, Y, active):
+    # Subtract 1 from Ys to account for Snorkel reserving the label 0 for abstains
+    # (but F.cross_entropy() expect 0-indexed labels)
```
Comment (ajratner) [inline snorkel/mtl/modules/utils.py]: Be clear that this is for hard (non-prob) labels

File: snorkel/mtl/trainer.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            train_split = [train_split]
+
+        train_dataloaders = [
+            dataloader for dataloader in dataloaders if dataloader.split in train_split
+        ]
+
+        if not train_dataloaders:
+            raise ValueError(
+                f"Cannot find the specified train_split "
+                f'{self.config["train_split"]} in datloaders.'
```
Comment (ajratner) [inline snorkel/mtl/trainer.py]: Typo
